provider/schema/execution_test.go

```diff
@@ -354,17 +349,18 @@ func TestExecutionData_ResolveTable(t *testing.T) {
 		mockDb.AssertNumberOfCalls(t, "CopyFrom", 1)
 		assert.Equal(t, expectedResource.data["name"], "other")
 		assert.Nil(t, err)
-		exec = NewExecutionData(mockDb, logger, testTable, false, nil, false)
-		mockDb.On("CopyFrom", mock.Anything, mock.Anything, false, mock.Anything).Return(nil)
+		exec = NewExecutionData(mockDb, logger, testTable, nil, false)
+		mockDb.On("CopyFrom", mock.Anything, mock.Anything, true, mock.Anything).Return(nil)
+		mockDb.On("RemoveStaleData", mock.Anything, testTable, exec.executionStart, mock.Anything).Return(nil)
 		_, err = exec.ResolveTable(context.Background(), mockedClient, nil)
-		mockDb.AssertNumberOfCalls(t, "Delete", 1)
+		mockDb.AssertNumberOfCalls(t, "RemoveStaleData", 2)
```

Comment:
did you check why this was called twice? feels wrong it should only be called once